tap_zendesk_chat/streams.py

```diff
@@ -463,7 +463,10 @@ def _sync(self, ctx: Context, full_sync: bool=False):
         next_url = None
         record_counts = []
 
-        while end_time < end_dt:
+        # should also force stop whenever start datetime is less than 30sec from current time
+        # (this causes a "requests less than 30 seconds old are not allowed error")
+        start_time_plus_30_seconds = start_dt + timedelta(seconds=30)
+        while end_time < end_dt and start_time_plus_30_seconds < end_dt:
```

Comment:
So IIUC, what this means is on a first successful run, we get a bookmark like `1600000`. On the next run, assuming that the current time is `1600001` (i.e 1 second after the last successful run), then we'll get an error. Is that right? Because otherwise `start_time_plus_30_seconds < end_dt` is always going to be true.

However, the error tells me `start_time` is the query param which is failing and not the initial request which has `start_dt`. This can only mean that the `next_url` they give us raises that error.

If so I would probably just opt to check the exception content and if it says the "30 seconds old" error, end the loop:
```python
while end_time < end_dt:
    try:
        response = self._request.get(...)
    except Zendesk400Exception as e:
        if e.description == 'Requests with a start_time ...':
            logger.info(...)
            break
```